oracle/datadog_checks/oracle/oracle.py

```diff
@@ -156,6 +157,16 @@ def _connection(self):
                     self.log.error("The JDBC connection failed with the following error: %s", str(e))
                     self._connection_errors += 1
             else:
+                if self._use_instant_client:
+                    try:
+                        oracledb.init_oracle_client()
+                    except oracledb.DatabaseError as e:
+                        self.log.warning('Oracle Instant Client is unavailable: %s', str(e))
+                        raise
```

Comment:
```suggestion
                        self._connection_errors += 1
```

Reply:
Do we want to increment the connection errors even though a connection isn't being started/created here?

Reply:
There are two possibilities I think. If instant client connection can't be made, we can either:
- Try with the native client instead as a backup, increment connection error, and continue with check
- Return no connection and increment connection error(similar to what we do with JDBC if that fails)

Reply:
Maybe I'm misunderstanding so correct me if I'm wrong, but the line:
```
oracledb.init_oracle_client()
```
isn't doing anything but making sure the Oracle libraries needed to run the Instant Client are present. It's not yet trying to establish a connect to the Oracle database yet which is why we still run:
```
self._cached_connection = self._oracle_connect()
```

Reply:
Yes that's correct, when I say `instant client connection can't be made`, I mean that the `init_oracle_client()` returns an exception. If this happens, we can either proceed with a regular non-instant client connection or return no connection.

Reply:
I think we should return no connection because the check cannot behave as the customer configured it. If they're trying to use thick mode because of the type of connection they want to establish, everyone is going to get lost looking at the connection errors down the line when the real issue is that the instant client is not set up correctly